.circleci/config.yml

```diff
@@ -44,12 +44,28 @@ jobs:
             for demo in !($IGNORED_DIRS); do
               if [ -d "$demo" ]; then
                 (
+                  # Call pyenv if we are using MacOS
+                  os="$(uname -s)"
+                  if [ "$os" == "Darwin" ]; then
+                    eval "$(pyenv init -)"
+                    pyenv local $PYTHON
+                  fi
+
+                  # Create demo specific virtual environments
                   env="envs/$demo"
                   python -m virtualenv "$env"
                   . "$env/bin/activate"
 
                   pip install -r "$demo/requirements.txt"
                   pip install coverage
+
+                  # Python 2.7 case
+                  # Note: using 2>&1 because python 2.7 sends version info to stderr
+                  python_version="$(python --version 2>&1)"
+                  if [[ "$python_version" =~ ^Python\ 2\.7\.[0-9]+$ ]]; then
+                    pip install mock
+                  fi
```

Comment:
This is unnecessary complicated and fragile.

If you need `mock` on Python 2 (only), include it in the requirements, together with the appropriate [environment marker (see PEP-496)](https://www.python.org/dev/peps/pep-0496/). See an [example in the cloud client](https://github.com/dwavesystems/dwave-cloud-client/blob/0364541298402a481562060759d3716abd220c6a/requirements.txt#L9).

Also, I believe there's no harm in unconditional install of `mock`, because it's designed to work in Python 2+3. At least until we drop support for Python 2.

Reply:
Moved mock to requirements.txt, following cloud client's example.